### PR TITLE
Introduce full RSpec testing

### PR DESCRIPTION
**Pull request by kmcelwee**

### CODE REVIEW

**Comment by kmcelwee** [comment]
Bitstream, Bundles, and Workflow objects cannot be created, so they weren't tested.

DSpace and Config were bug-heavy, and complicated to implement when testing in environment. Put off for now.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# require 'ruby-debug'
```
Comment (jrgriffiniii) [inline spec/spec_helper.rb:1]: 👍

File: spec/lib/dspace/dspace_spec.rb [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    before do 
+      # TODO: Reset DSpace before each load
+      #   DSpace.context.abort is a good start, but variables need to be reset as well
+    end
+
+    after :all do
+      # TODO: Make sure DSpace is loaded properly for all tests
+    end
+
+=begin
```
Comment (jrgriffiniii) [inline spec/lib/dspace/dspace_spec.rb]: While this is syntactically valid, this is a fault of mine given that I don't have RuboCop enforcing certain style rules. If I recall correctly, DRDS uses a style rule which prefers blocks of comments to this.

**Review by jrgriffiniii — APPROVED** [review]

**Comment by jrgriffiniii** [comment]
This needs a rebase with `.gitignore`, and unfortunately, once I modify the branch, I will need a third party to review this. I apologize for the inconvenience, but thank you very much for completing this!

File: spec/lib/dspace/dspace_spec.rb (hunk 2 above)
Comment (kmcelwee) [reply spec/lib/dspace/dspace_spec.rb]: Fixed in latest commit. Should I open an issue to setup Rubocop in `dspace-cli` and `dspace-jruby`?

**Comment by jrgriffiniii** [comment]
As I cannot address this without rebasing the branch, I'm going to set this as a draft.

**Pull request closed without merging**